[pilot] app_version and app_build should not be fetched from a local IPA or PKG when distribute_only is set

tremblay opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
🔑
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Resolves #20479

### Description
<!-- Describe your changes in detail. -->
<!-- Please describe in detail how you tested your changes. -->

As mentioned in the title and the linked issue, a local IPA or PKG should not override the info provided to the command if `distribute_only` is set.

My particular test case used a repo with an iOS and macOS target, and an older macOS `.pkg` file sitting in the repo. When I ran the command from the linked issue, I ended up waiting for an iOS build with the version number on the old macOS `.pkg`, even though I passed the correct `build_number` to the command. With this change, the `distribute_only` argument will cause the code to fall through to the step where `build_number` is picked up from the `config`.

### Testing Steps
<!-- Optional: steps, commands, or code used to test your changes. -->
<!-- Providing these will reduce the time needed for testing and review by the fastlane team. -->

tremblay committed:
app_version and app_build should not be fetched from a local IPA or PKG when distribute_only is set

tremblay commented:
Unfortunately, I'm not seeing what I've done wrong with regards to the CircleCI checks

getaaron commented:
@tremblay can you please connect circleci to your github account and rerun the build? once you've logged into CircleCI using your GitHub account, you should be able to rerun this branch from the CircleCI UI (or just push an empty commit to this branch if that's easier)

getaaron reviewed (commented):
I think it'd be helpful to add a test, you could basically copy this one:

https://github.com/fastlane/fastlane/blob/c6ef9b163ac09865cb9be80d9da6216a6d763c82/pilot/spec/build_manager_spec.rb#L453-L469

but pass in a pkg, build number, and distribute_only, then `expect` that it uses the number you pass in and it does _not_ call `PkgFileAnalyser`

WDYT?

tremblay committed:
add a test for the new behavior, remove a duplicate test

tremblay commented:
@getaaron thanks for the help! CircleCI is now running, and I added a test like you suggested. I also noticed that there were 2 copies of the exact same test `wait given :app_version and :build_number`, so I removed the 2nd one. Hope that's alright!

getaaron reviewed (approved):
Awesome, thanks so much for your work on this! 🚀🚀🚀
